Content/NPCs/Moonstone/DreambeastSystems.cs

```diff
@@ -0,0 +1,112 @@
+﻿using StarlightRiver.Content.Biomes;
+using StarlightRiver.Core.Systems.MetaballSystem;
+using System.Linq;
+using Terraria.ID;
+using Terraria.Graphics.Effects;
+
+namespace StarlightRiver.Content.NPCs.Moonstone
+{
+
+	internal class DreamBeastDrawSystem : ModSystem
+	{
+		public override void Load()
+		{
+			On_Main.DrawNPCs += DrawDreamBeast;
+		}
+
+		private void DrawDreamBeast(On_Main.orig_DrawNPCs orig, Main self, bool behindTiles)
+		{
+			orig(self, behindTiles);
+
+			if (!behindTiles)
+				DreamBeastActor.DrawSpecial();
+		}
+	}
+
+	internal class DreamBeastActor : MetaballActor
+	{
+		public NPC activeBeast;
+
+		public override bool Active => NPC.AnyNPCs(ModContent.NPCType<Dreambeast>());
+
+		public override Color OutlineColor => new Color(220, 200, 255) * (activeBeast?.Opacity ?? 0) * 0.5f;
+
+		public override void DrawShapes(SpriteBatch spriteBatch)
+		{
+			for (int k = 0; k < Main.maxNPCs; k++)
+			{
+				NPC npc = Main.npc[k];
+
+				if (npc.ModNPC is Dreambeast)
+					(npc.ModNPC as Dreambeast).DrawToMetaballs(spriteBatch);
+			}
+		}
+
+		public override bool PostDraw(SpriteBatch spriteBatch, Texture2D target)
+		{
+			activeBeast = Main.npc.FirstOrDefault(n => n.active && n.type == ModContent.NPCType<Dreambeast>()); //TODO: proper find for onscreen beast
+
+			return false;
+		}
+
+		public static void DrawSpecial()
+		{
+			MetaballSystem.actorsSem.WaitOne();
+
+			NPC activeBeast = Main.npc.FirstOrDefault(n => n.active && n.type == ModContent.NPCType<Dreambeast>());
+
+			if (activeBeast is null)
+			{
+				MetaballSystem.actorsSem.Release();
+				return;
+			}
+
+			Texture2D target = MetaballSystem.actors.FirstOrDefault(n => n is DreamBeastActor).Target.RenderTarget;
+
+			float lunacy = Main.LocalPlayer.GetModPlayer<LunacyPlayer>().lunacy;
+			if (lunacy >= 0 && lunacy < 100)
```

Comment:
Design question: If the range for this is 0-100, why not have it be a float instead of an integer? It would be simpler to multiply effects

Reply:
Was honestly just a bit easier to work with since some of the numbers for passive lunacy gain was actually already quite small.